### DISCUSSION TITLE
Is there any ways to delete Custom Xcom values when deleteting a dag through the Airflow UI

### DISCUSSION THREAD

**Discussion by kimminw00** [apache/airflow#31493] [Q&A] [answered]
I override serialize_value & deserialize_value methods to save xcom values to S3. 
But when deleteting a dag through the Airflow UI, S3 still has xcom values.
Is there any ways to delete Custom Xcom values?

**Comment by zoid-w** [comment]
If you are searching for a way to clear the xcom values when rerunning a dagrun, this might help.
You should rewrite the clear method in the xcom_backend.
You can find some information [here](https://docs.astronomer.io/learn/xcom-backend-tutorial). 

Besides that I do not think there is an option to do something similar when deleting a DAG. You could ofcourse create a pipeline that screens your dagbag and finds the structure within S3 to do a cleanup. Or ensure a time-based cleanup of the bucket if that does not conflict with other requirements for you.

**Reply by potiuk** [reply, marked as answer]
Nope. there is no such functionality. You need to make such cleanups on your own - for example you can run a cleanup script that wil read (Via API) which dags are available and delete all other's stored data. Should be easy - you even have Python client  thatn can help writing appropriate script: https://pypi.org/project/apache-airflow-client/
